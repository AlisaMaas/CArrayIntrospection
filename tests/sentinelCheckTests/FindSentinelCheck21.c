/**
 * This check tests we correctly do not mix up array
 * arguments across functions.
 *
 * We expect to find one non-optional sentinel check for string 
 * and 1 for string2 in both foo and find.
 **/

void printc(char c) {}
int foo(char string[], char string2[], int distance) {
	for (int i = 0; i < distance; i++) {
		if (string[i] == '\0') {
			break;
		}
		if (string2[i] == '\0')
			break;
	}
	return 1;
}
int find(char string[], char string2[], int distance) {
	for (int i = 0; i < distance; i++) {
		if (string[i] == '\0') {
			break;
		}
		if (string2[i] == '\0')
			break;
	}
	return 1;
}
