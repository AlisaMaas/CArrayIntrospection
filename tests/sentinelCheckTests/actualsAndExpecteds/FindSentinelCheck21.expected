Printing analysis 'Promote Memory to Register' for function 'printc':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'foo':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'find':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Find each branch used to exit a loop when a sentinel value is found in an array' for function 'printc':
Analyzing function: printc
	We found: 0 loops
Printing analysis 'Find each branch used to exit a loop when a sentinel value is found in an array' for function 'foo':
Analyzing function: foo
	We found: 1 loops
	Examining string in loop for.cond
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body
	Examining string2 in loop for.cond
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			if.end
Printing analysis 'Find each branch used to exit a loop when a sentinel value is found in an array' for function 'find':
Analyzing function: find
	We found: 1 loops
	Examining string in loop for.cond
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body
	Examining string2 in loop for.cond
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			if.end
